### DISCUSSION TITLE
GraphQL mismatch for seemingly no reason

### DISCUSSION THREAD

**Discussion by emilyrbowe** [tinacms/tinacms#3896] [Help] [answered]
As of this AM (~9:40am ET), our Tina site has been giving us the following error: `GraphQL Schema Mismatch. Editing may not work. If you just switched branches, try going back to the previous branch`

<img width="456" alt="image" src="https://user-images.githubusercontent.com/7975988/235786321-546433d6-5213-4dea-97c4-bc83e49e2cde.png">

We did not switch branches and none of the commits that happened within the Tina editor had anything to do with the config. Any ideas how to solve this or if there was something from the TinaCMS end that might have contributed to this?

**Comment by kldavis4** [comment]
@emilyrbowe what's the client id for the project?

**Reply by emilyrbowe** [reply]
`097f9d05-d307-4978-823b-d332ea55d27e` is the client ID for the project.

**Reply by emilyrbowe** [reply]
For additional context, all seems to be okay when working in dev mode, but when we try to build our site on Netlify, we are getting this error during the build step:
![image](https://user-images.githubusercontent.com/7975988/235795094-b68d08a5-a2f6-49eb-8ec9-873f396e5ab6.png)

**Reply by kldavis4** [reply]
I don't see any schema change events from your github repository. This issue occurs when the `tina/tina-lock.json` in your GitHub repository is different than the one that the TinaCMS `/admin` was built with. Were there any recent updates to your deployed site (upgraded @tinacms/cli and/or tinacms)?

**Reply by kldavis4** [reply]
when you run in dev mode locally, are there any local changes to the `tina/tina-lock.json`?

**Reply by emilyrbowe** [reply]
Didn't upgrade either @tinacms/cli or tinacms. When we first started with our migration from Forestry, our site had a `master` branch and we used the opportunity to create a `main` branch instead, which is what it builds from now.

**Reply by emilyrbowe** [reply]
And no changes in dev mode locally.

**Reply by kldavis4** [reply]
I see your repository is public - taking a look now

**Reply by kldavis4** [reply, marked as answer]
There's definitely a difference between what's in tina cloud and your repository. Go to your project configuration page in tina cloud and click the reindex button on the `main` branch. That should sync tina cloud with what is in your repository (should normally happen automatically).

You should make sure and commit the package-lock.json or yarn-lock.json (depending on whether you are using npm or yarn). I believe that not being there might be a contributing factor since it controls what specific version of tina will be installed in your CI.

**Reply by kldavis4** [reply]
Also when I run the `dev` command on the `main` branch, it does cause changes in your `tina/tina-lock.json`. Those changes need to be committed and pushed up to your repository.

**Reply by kldavis4** [reply]
Also noticed you commented about the tina/.gitignore line for the __generated__ folder. You don't need that folder committed, so you can uncomment that.

**Reply by emilyrbowe** [reply]
Ah, okay. Thanks — I will do that and hope that it helps fix it. I reindexed the project earlier today with no change, but perhaps after the tweaks this afternoon it will fix it.

**Reply by emilyrbowe** [reply]
Committing the package-lock.json seems to have done the trick — as well as whatever changes seemed to be hiding in the tina-lock.json. Thank you, @kldavis4!
